scripts/compactor_runner.py

```diff
@@ -282,17 +244,71 @@ def _run_corfu_compactor(self):
+                        and server_address["NetworkInterface"]:
+                    self._print_and_log("WARNING: _run_corfu_compactor: Couldn't load the hostname from args or config "
+                                        "file. Please check the parameters. Continuing self IP discovery with {}"
+                                        " interface and {} version.".format(server_address["NetworkInterface"],
+                                                                            server_address["NetworkInterfaceVersion"]))
+                    self._config.hostname = self._resolve_ip_address(
+                        network_interface=server_address["NetworkInterface"],
+                        network_interface_version=server_address["NetworkInterfaceVersion"]
+                    )
+                else:
+                    raise RuntimeError("_run_corfu_compactor: Couldn't load the hostname or network interface info"
+                                       " from args or config file. Please check the configurations.")
+            else:
+                # When server address param is not there in the config file or hostname is not present in the args
+                raise RuntimeError("_run_corfu_compactor: Couldn't load the server_address from args or config file."
+                                   " Please check the configurations.")
+
+        if not self._config.corfu_port and server_address and server_address["Port"]:
+            self._config.corfu_port = str(server_address["Port"])
+        else:
+            # When port param is not there in the config file or port is not present in the server args
+            raise RuntimeError("_run_corfu_compactor: Couldn't load the port from args or config file."
+                               " Please check the configurations.")
+
+        self._print_and_log("_run_corfu_compactor: Configured compacter hostname as "
+                            + self._config.hostname
+                            + ", port as "
+                            + self._config.corfu_port)
+
         if self._config.startCheckpointing:
-            grep_running_tool = "ps aux | grep 'python3 /usr/share/corfu/scripts/compactor_runner.py' | grep 'startCheckpointing' | grep -v 'grep' | grep " + self._config.corfu_port
 
             try:
+                grep_running_tool = "ps aux | grep 'python3 /usr/share/corfu/scripts/compactor_runner.py' | grep 'startCheckpointing' | grep -v 'grep' | grep " + self._config.corfu_port
                 grep_tool_result = check_output(grep_running_tool, shell=True).decode()
+                self._print_and_log("Result for " + grep_running_tool + ":\n" + grep_tool_result)
             except Exception as ex:
-                self._print_and_log("Failed to run grep command: " + grep_running_tool + ", error: " + str(ex))
+                self._print_and_log("Failed to run grep_running_tool command. Error: " + str(ex))
```

Comment:
Better to keep the existing print line. Having the grep_running_tool line printed has been useful for debugging in the past

Reply:
Done, Thank you.